docs/source/operations/operations_computation.rst

```diff
@@ -0,0 +1,741 @@
+operations between those CRS in the EPSG registry itself, transformations from
+other authorities will be ignored (except if there are in the PROJ authority,
+which can be used as an override).
+
+As those results all involve operations that have not perfect accuracy and that
+there are of use does not cover the area of use of the 2 CRSs, a
+'Ballpark geographic offset from NAD27 to NAD83' operation is synthetized by PROJ.
+This operation is a rather dummy operation that only takes into account potential
+difference of axis orders (long-lat vs lat-long), units (degree vs grads) and
+prime meridian (Greewich vs Paris/Rome/other historic prime meridians). It does
+not attempt any datum shift, hence the "ballpark" qualifier in its name.
+
+Filtering and sorting of coordinate operations
+----------------------------------------------
+
+The last step is to filter and sort results in a relevance order.
+
+The filtering
+takes into account a minimum accuracy that the user might have expressed, an
+area of use on which the coordinate operation(s) must apply and if the absence
+of grids needed by some operations must result in discarding the operations.
+
+The sorting algorithm is a set of heuristics. It is implement by the `operator ()`
+method of the SortFunction structure.
+The following criteria are used in the following order (first listed, first applied):
+
+* put at top operations that can be expressed as a PROJ instanciable operation
+  (the database might list operations whose method is not (yet) implemented by PROJ)
+* put at top operations that do not have a synthetic ballpark vertical transformation
+  (occurs when there is a geoid model)
+* put at top operations that do not have a synthetic ballpark horizontal tranformation
+* put at top operations that refer to shift grids that are locally available
+* put at top operations that refer to grids that are available in one of the proj-datumgrid
+  packages, but not necessarily locally available
+* put at top operations that have a known accuracy
+* if two operations have unknown accuracy, then put at top the one that use grid
+  if the other one does not (grid based operations are assumed to be more precise
+  than operations relying on a few parameters)
+* put at top operations whose area of use is larger (note: the computation of the
+  are of use is approximate, based on a bounding box)
+* put at top operations that have a better accuracy
+* in case of same accuracy, put at top operations that do not use grids (operations
+  that use only parameters will be faster)
+* put at top operations that involve less transformation steps
+* and for completness, if two operations are comparable given all the above criteria,
+  put at first the one that has the shorter name, and if they have the same lengt, sort
+  by lexicographic order (obviously completely arbitrary, but a sorting
+  algorithm must be able to compare all entries)
```

Comment:
I am confused by this. I know this is not easy to express this simply! You are building the list from the bottom, yes? The second and third step feels wrong to me, that you put operations without ballpark operations almost at the bottom at the list. I would intiutively assume that operations like that would be more direct and thus better suited at the top of the list?

At the moment I don't have a good suggestion for how to improve this.

Reply:
> You are building the list from the bottom, yes?

not sure to understand what you mean...

> The second and third step feels wrong to me, that you put operations without ballpark operations almost at the bottom at the list

So I wrote "put at top operations that do not have a synthetic ballpark vertical transformation" . Maybe you missed the negation ? Or "put at top" is not clear ? "Put at top" = "more relevant" (the one you see first in the output of projinfo)

Reply:
Yes, I think it is the "put a top" that is putting me off, so to speak. As well as there being many steps of which many are not trivial. I feel like I need a figure with a decision tree to decipher this algortithm :-)

If I understand it correctly, you start by making a list of all possible operations and then narrowing it down according to various "test" of subsequent priority. The final list is sorted by the number "tests" that the operations passes, with the first operation in the list being the one with the most passed "tests". Have I understood that somewhat correctly?

Reply:
Not sure I understand your questions... lol
Let me try to put it otherwise. So this long enumeration is about the transcription of https://github.com/OSGeo/PROJ/blob/master/src/iso19111/coordinateoperation.cpp#L10777 . This is just a sorting function. It doesn't remove operations (operation filtering happens mostly before sorting, and also a bit after, but I didn't even get into that detail). The sort function takes 2 operations, a and b, and determines which one must be considered as "less" than the other. If it returns true, which means a < b, then a will be at the top of the list, and at the top of the output of projinfo, that is the more relevant.

Reply:
@kbevers I've tried in fa7633e to clarify a bit this sorting stuff

Reply:
Thanks, this is a much better description of the process. I had to read the code to understand the other, I think this one can stand by itself. Good job!